src/components/PuzzleGrid/index.jsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState, useEffect } from "react";
+import { Col, Row } from "antd";
+import puzzledImage from "./../../images/habitat.jpg";
+import shuffle from "./../memorygame/shuffle";
+import "./index.css";
+
+const NUM_OF_TILES = 6;
+// PuzzleGrid is the component that generates and includes the tiles of the puzzle
+// it generates 6 tiles ordered vertically on top of each other
+// each tile's background is 1/6 of the image. After the tiles are generated they are put
+// in an array and ordered randomly
+// a user can click on one tile to select and then click on another tile to swap their places
+// the user will swap the places of each tile to put it in its correct position
+// when all tiles are placed in their correct position the full image is displayed
+const PuzzleGrid = () => {
+  const [currentPuzzleState, setCurrentPuzzleState] = useState([]);
+  // selectedTile state shows if there is a tile selected and which one is selected if any
+  const [selectedTile, setSelectedTile] = useState(null);
+  // if unsolved it displays the tiles and when solved it displays the full image
+  const [isSolved, setIsSolved] = useState(false);
+
+  const tilesGenerator = () => {
+    const newTilesArray = [];
+    // for loop to generate tiles where tiles are puzzle pieces
+    for (let tileId = 0; tileId < NUM_OF_TILES; tileId++) {
+      let tileTopStyling = tileId * 100;
+      // value of selected is used for styling purposes where the selected tile should be highlighted
+      // value of top is used for fitting the image inside the tile
+      let newTile = { id: tileId, selected: false, top: tileTopStyling };
+      newTilesArray.push(newTile);
+    }
+
+    const randomizedTilesArray = shuffle(newTilesArray);
+    // Set the randomized array as generated tiles
+    setCurrentPuzzleState(randomizedTilesArray);
+  };
+
+  useEffect(() => {
+    tilesGenerator();
+  }, []);
+
+  useEffect(() => {
+    checkSolution();
+  }, [currentPuzzleState]);
+
+  const handleSelect = (clickedTileId) => {
+    // if selectedTile's value is not equal to null (meaning there a tile seleted already)
+    // then when another tile is clicked swap both tiles' places
+    // else set the selectedTile to the id of the clicked tile
+    if (selectedTile !== null) {
+      const prevId = selectedTile;
```

Comment:
How can a tile be id

Reply:
I don't understand this comment

Reply:
`const prevId = selectedTile;`    how can a tile be equal to id

Reply:
you have a variable called prevId and you assign it to selectedTile. the word tile implies that its an object. an id implies its a string

this is a good one to be careful of too, what you mean is selectedTileId

Reply:
PTAL